docs/ARCHITECTURE.md

```diff
@@ -201,3 +201,67 @@ can be found under various directories such as `factory/`, `modprobe.d/`, `netwo
+image to allow loading it as a UEFI executable, `systemd-stub`. These components fully support UEFI Secure
+Boot, and this paragraph will describe their security posture and how they comply with industry-standard
+expectations for UEFI Secure Boot workflows.
+
+### Core Design
+The core design goals for these components is simplicity, achieved by offloading security-critical tasks to
+the firmware or earlier stages of the boot process (e.g.: `Shim`).
+
+When Secure Boot is enabled, these components are designed to avoid executing or loading or using
+unauthenticated payloads that could compromise the boot process, with special care taken for anything that
+could affect the system before `ExitBootServices()` has been called. For example, when additional resources
+are loaded, if running with Secure Boot enabled, they will be validated before use. The only exceptions are
+the bootloader's own textual configuration files. There are no build time or runtime configuration options
+that can be set to weaken the security model of these components when Secure Boot is enabled.
+
+The role of `systemd-boot` is to discover next stage components in the ESP (and XBOOTLDR if present), via
+filesystem enumeration or explicit configuration files, and present a menu to the user, to choose the next
+step. This auto discovery mechanism is described in details in the [BLS (Boot Loader
+Specification)](https://uapi-group.org/specifications/specs/boot_loader_specification/).
+
+The role of `systemd-stub` is to load and measure in the TPM the post-bootloader stages, such as the kernel,
+initrd and kernel command line. These payloads can be augmented, and such augmentations are measured too.
+
+Various EFI variables, under the vendor UUID `4a67b082-0a4c-41cf-b6c7-440b29bb8c4f`, are set and read by
+these components, to pass metadata and configuration between different stages of the boot process, as
+defined in the [Boot Loader Interface](https://systemd.io/BOOT_LOADER_INTERFACE/).
+
+### Dependencies
+Neither of these components implements cryptographic primitives, cryptographic checks or file system drivers.
+File access to the ESP is implemented solely via the appropriate UEFI file protocols. Verification of next
+stage payloads is implementend solely via the appropriate UEFI image load protocols, which means
+authenticode signature checks are again done by the firmware or `Shim`. As a consequence, no external
+security-critical libraries (such as OpenSSL or gnu-efi) are used, linked or embedded.
+
+### Additional Resources
+BLS Type #1 entries allow the user to load two types of additional resources, kernel command line arguments
+and Devicetree blobs, that are not validated before use, as they do not carry signatures. For this reason,
+when Secure Boot is enabled, loading these resources is automatically disabled. There is no override for
+this security mechanism, neither at build time nor at runtime.
```

Comment:
well, initrds are also loaded and not authenticated. this is the old model after all.